client/components/seo/meta-title-editor/mappings.js

```diff
@@ -0,0 +1,41 @@
+// Maps between different title format formats
+//
+// raw is a string like: '%site_name% | %tagline%'
+//                        \_________/   \_______/
+//                             |            |
+//                             \------------\- tokens
+//
+// native is an array of plain-text strings and tokens
+//   [ { type: 'postTitle' }, { type: 'string', value: ' on ' }, { type: 'siteName' } ]
+//     \___________________/                                     \__________________/
+//               |                                                  |
+//               \--------------------------------------------------\--- tokens
+//
+//   @see README for `TokenField`
+//   [
+//   	{ value: 'Site Name', type: 'siteName' },
+//   	{ value: ' | ', type: 'string' },
+//   	{ value: 'Post Title', type: 'postTitle' }
+//   ]
+//
+
+import camelCase from 'lodash/camelCase';
+import join from 'lodash/join';
+import map from 'lodash/map';
+import snakeCase from 'lodash/snakeCase';
+import split from 'lodash/split';
+
+export const removeBlanks = l => l.filter( ( { value } ) => '' !== value );
+
+const placeholderPattern = /(%[a-zA-Z_]+%)/;
```

Comment:
it's currently waiting validation, the first point in the remaining tasks in the PR description. that should flag an error to the user